db/migrate/20250212192052_add_constraints_to_user_action_events.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class AddConstraintsToUserActionEvents < ActiveRecord::Migration[7.2]
+  def change
+    safety_assured do
+      # Add indexes
+      add_index :user_action_events, :event_id, unique: true
+      add_index :user_action_events, :event_type
+
+      # Add NOT NULL constraints
```

Comment:
I think you're good to remove this comment too

Reply:
Same here.